mono_repo/lib/src/presubmit_command.dart

```diff
@@ -0,0 +1,128 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:io/ansi.dart';
+import 'package:path/path.dart' as p;
+import 'package:pub_semver/pub_semver.dart';
+
+import 'travis_config.dart';
+import 'utils.dart';
+
+class PresubmitCommand extends Command {
+  @override
+  String get name => 'presubmit';
+
+  @override
+  String get description => 'Run the travis presubmits locally.';
+
+  PresubmitCommand() {
+    argParser.addOption('package',
+        help: 'The package(s) to run on, defaults to all packages',
+        allowMultiple: true,
+        abbr: 'p');
+    argParser.addOption('task',
+        help: 'The task(s) to run, defaults to all tasks',
+        allowMultiple: true,
+        abbr: 't');
+    argParser.addOption('sdk',
+        help: 'Which sdk to use for match tasks, defaults to current sdk',
+        allowMultiple: false,
+        defaultsTo: _currentSdk,
+        allowed: ['dev', 'stable']);
+  }
+
+  @override
+  Future run() async {
+    var passed = await presubmit(
+        packages: argResults['package'] as List<String>,
+        tasks: argResults['task'] as List<String>,
+        sdkToRun: argResults['sdk'] as String);
+
+    // Set a bad exit code if it failed.
+    if (!passed) exitCode = 1;
+  }
+}
+
+final _currentSdk =
+    new Version.parse(Platform.version.split(' ').first).isPreRelease
+        ? 'dev'
+        : 'stable';
+
+/// A temp dir to dump logs to for errors.
+Directory __tmpDir;
+Directory get _tmpDir => __tmpDir ??= Directory.systemTemp.createTempSync();
+
+Future<bool> presubmit(
+    {Iterable<String> packages,
+    Iterable<String> tasks,
+    String sdkToRun,
```

Comment:
Should have a TODO here about SDK stable vs dev.

If someone was running a 1.24.0-dev SDK, I'm guessing that'd show up as 'dev', right?

stable/dev are stable for the time being, but in theory they are moving targets.

We could get crazy and actually hit our download server and check what current dev/stable are. But that seems a lot of work for now.

A TODO note would be nice, though

Reply:
Ya it would, I am not sure the best way to handle this in general... we could just run all tasks using the current sdk and forget about the whole dev/stable stuff...